scripts/execute_cypress_v2.sh

```diff
@@ -0,0 +1,188 @@
+    exit 1
+  fi
+
+  local connector="$1"
+  local service="$2"
+  local tmp_file="$3"
+
+  print_color "yellow" "Executing tests for ${service} with connector ${connector}..."
+
+  export REPORT_NAME="${service}_${connector}_report"
+
+  if ! CYPRESS_CONNECTOR="$connector" npm run "cypress:$service"; then
+    echo "${service}-${connector}" >> "${tmp_file}"
+  fi
+}
+export -f execute_test
+
+# Function to run tests
+function run_tests() {
+  local jobs="${1:-1}"
+  tmp_file=$(mktemp)
+
+  # Ensure temporary file is removed on script exit
+  trap 'cleanup' EXIT
+
+  for service in "${connector_map[@]}"; do
+    declare -n connectors="$service"
+
+    if [[ ${#connectors[@]} -eq 0 ]]; then
+      # Service-level test (e.g., payment-method-list or routing)
+      [[ $service == "payment_method_list" ]] && service="payment-method-list"
+
+      echo "Running ${service} tests without connectors..."
+      export REPORT_NAME="${service}_report"
+
+      if ! npm run "cypress:${service}"; then
+        echo "${service}" >> "${tmp_file}"
+      fi
+    else
+      # Connector-specific tests (e.g., payments or payouts)
+      print_color "yellow" "Running tests for service: '${service}' with connectors: [${connectors[*]}] in batches of ${jobs}..."
+
+      # Execute tests in parallel
+      printf '%s\n' "${connectors[@]}" | parallel --jobs "${jobs}" execute_test {} "${service}" "${tmp_file}"
+    fi
+  done
+
+  # Collect failed connectors
+  if [[ -s "${tmp_file}" ]]; then
+    failed_connectors=($(< "${tmp_file}"))
+    print_color "red" "One or more connectors failed to run:"
+    printf '%s\n' "${failed_connectors[@]}"
+    exit 1
+  else
+    print_color "green" "Cypress tests execution successful!"
+  fi
+}
+
+# Function to check and install dependencies
+function check_dependencies() {
+  # parallel and npm are mandatory dependencies. exit the script if not found.
+  local dependencies=("parallel" "npm")
+
+  for cmd in "${dependencies[@]}"; do
+    if ! command_exists "$cmd"; then
+      print_color "red" "ERROR: ${cmd^} is not installed!"
+      exit 1
+    else
+      print_color "green" "${cmd^} is installed already!"
+
+      if [[ ${cmd} == "npm" ]]; then
+        npm ci || {
+          print_color "red" "Command \`npm ci\` failed!"
+          exit 1
+        }
+      fi
+    fi
+  done
+}
+
+# Cleanup function to handle exit
+function cleanup() {
+  print_color "yellow" "Cleaning up..."
+  if [[ -d "cypress-tests" ]]; then
+    cd -
+  fi
+
+  if [[ -n "${tmp_file}" && -f "${tmp_file}" ]]; then
+    rm -f "${tmp_file}"
+  fi
+}
+
+# Main function
+function main() {
+  local command="${1:-}"
+  local jobs="${2:-5}"
+
+  # Ensure script runs from 'cypress-tests' directory
+  if [[ "$(basename "$PWD")" != "cypress-tests-v2" ]]; then
+    print_color "yellow" "Changing directory to 'cypress-tests-v2'..."
+    cd cypress-tests-v2 || {
+      print_color "red" "ERROR: Directory 'cypress-tests-v2' not found!"
```

Comment:
the only thing that is different here is this part of code when compared to `execute_cypress.sh`.  

i feel, it is better to pass the directory name as an argument from ci (while keeping `cypress-tests` as default) to `execute_cypress.sh` instead of having this file.

Reply:
If everything else in the script remains same, then sure, we can have the directory accepted as a command line arg.

Reply:
i did verify. the difference that exist in the entire file is this section of code.